tests/unit/models/test_model_interfaces.py

```diff
@@ -586,6 +648,38 @@ def test_vgp_update_updates_num_data() -> None:
     assert new_num_data - num_data == 2
 
 
+def test_vgp_update() -> None:
+    x = tf.constant(np.arange(5).reshape(-1, 1), dtype=gpflow.default_float())
+
+    data = Dataset(x, _3x_plus_10(x))
+    m = VariationalGaussianProcess(_vgp(data.query_points, data.observations))
+
+    x_new = tf.concat([x, tf.constant([[10.0], [11.0]], dtype=gpflow.default_float())], 0)
+    new_data = Dataset(x_new, _3x_plus_10(x_new))
+
+    reference_model = _vgp(data.query_points, data.observations)
+    reference_model_new = _vgp(new_data.query_points, new_data.observations)
+
+    internal_model = m.model
```

Comment:
a bit nitpicky, but its too verbose with all these variable assignments - everything can be called from within allclose function in relatively succinct fashion

Reply:
Hopefully better now!